### PR TITLE
operator: prepare for publishing at operatorhub.io

### PR DESCRIPTION
**Pull request by rojkov**
This PR adds `packagemanifests` folder which can be regenerated with `make packagemanifests` command and is supposed to be uploaded as is (using Drag-n-Drop) to operatorhub.io -> Contribute -> Package Your Operator to make a "publishable" package.

Contibutes to #385

### CODE REVIEW

**Comment by rojkov** [comment]
I've updated the CRD descriptions to make them a bit clearer when they end up on the Operator's web page at the operatorhub.

File: DEVEL.md [hunk 1]
```diff
@@ -162,4 +162,39 @@ $ make generate
...
+$ operator-sdk olm uninstall
+```
+
+Review the package manifests by uploading the generated `packagemanifests` folder to
+https://operatorhub.io -> Contribute -> Package Your Operator.
+
+Clone the [Community Operators](https://github.com/operator-framework/community-operators) repo:
+```
+$ git clone https://github.com/operator-framework/community-operators
+```
```
Comment (mythi) [inline DEVEL.md:193]: Is this an additional step suggested? I don't see pmem-csi there.

File: pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go [hunk 2]
```diff
@@ -71,8 +71,11 @@ type FpgaDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="FPGA Device Plugin"
 
-// FpgaDevicePlugin is the Schema for the fpgadeviceplugins API.
+// FpgaDevicePlugin is the Schema for the fpgadeviceplugins API. It represents
+// the FPGA device plugin responsible for advertising FPGA hardware resources to
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go:77]: how about "Intel FPGA hardware resource"...

File: pkg/apis/deviceplugin/v1/gpudeviceplugin_types.go [hunk 3]
```diff
@@ -72,8 +72,11 @@ type GpuDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="GPU Device Plugin"
 
-// GpuDevicePlugin is the Schema for the gpudeviceplugins API.
+// GpuDevicePlugin is the Schema for the gpudeviceplugins API. It represents
+// the GPU device plugin responsible for advertising GPU hardware resources to
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/gpudeviceplugin_types.go:78]: how about "Intel GPU hardware resource"...

File: pkg/apis/deviceplugin/v1/qatdeviceplugin_types.go [hunk 4]
```diff
@@ -81,8 +81,11 @@ type QatDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="QAT Device Plugin"
 
-// QatDevicePlugin is the Schema for the qatdeviceplugins API.
+// QatDevicePlugin is the Schema for the qatdeviceplugins API. It represents
+// the QAT device plugin responsible for advertising Intel QuickAssist hardware resources to
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/qatdeviceplugin_types.go:87]: how about "Intel QuickAssist Technology hardware resources"...

File: pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go [hunk 5]
```diff
@@ -71,8 +71,11 @@ type FpgaDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="FPGA Device Plugin"
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go:74]: for all `displayName` I'd add Intel where applicable and write out the acronym (see QAT/SGX below)... or is there a character limit?

File: pkg/apis/deviceplugin/v1/sgxdeviceplugin_types.go [hunk 6]
```diff
@@ -75,8 +75,11 @@ type SgxDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="SGX Device Plugin"
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/sgxdeviceplugin_types.go:78]: here, let's spell out SGX: Intel Software Guard Extensions Device Plugin

File: pkg/apis/deviceplugin/v1/sgxdeviceplugin_types.go [hunk 7]
```diff
@@ -75,8 +75,11 @@ type SgxDevicePluginStatus struct {
...
 
-// SgxDevicePlugin is the Schema for the sgxdeviceplugins API.
+// SgxDevicePlugin is the Schema for the sgxdeviceplugins API. It represents
+// the SGX device plugin responsible for advertising SGX hardware resources to
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/sgxdeviceplugin_types.go:81]: s/hardware/device node/

File: pkg/apis/deviceplugin/v1/qatdeviceplugin_types.go [hunk 8]
```diff
@@ -81,8 +81,11 @@ type QatDevicePluginStatus struct {
 // +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
 // +kubebuilder:printcolumn:name="Node Selector",type=string,JSONPath=`.spec.nodeSelector`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="QAT Device Plugin"
```
Comment (mythi) [inline pkg/apis/deviceplugin/v1/qatdeviceplugin_types.go:84]: Intel QuickAssist Technology Device Plugin

**Review by mythi — COMMENTED** [review]
just some small wording related comments. Looks good otherwise!

File: DEVEL.md (hunk 1 above)
Comment (rojkov) [reply DEVEL.md:193]: It's over here: https://github.com/operator-framework/community-operators/tree/master/upstream-community-operators/pmem-csi-operator

File: pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go (hunk 5 above)
Comment (rojkov) [reply pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go]: I've just checked the preview page: even if there is a limit all the display names fit still it. Though the description fades away after "...device plugin responsible for...". Not sure it's a problem though.

File: pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go (hunk 2 above)
Comment (rojkov) [reply pkg/apis/deviceplugin/v1/fpgadeviceplugin_types.go]: Updated

**Review by mythi — APPROVED** [review]

**Pull request merged**